packages/create-toolpad-app/src/index.ts

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env node
+
+import chalk from 'chalk';
+import { execaCommand } from 'execa';
+import inquirer from 'inquirer';
+import * as fs from 'fs/promises';
+import path from 'path';
+
+type PackageManager = 'npm' | 'pnpm' | 'yarn';
+
+function getPackageManager(): PackageManager {
+  const userAgent = process.env.npm_config_user_agent;
+
+  if (userAgent) {
+    if (userAgent.startsWith('yarn')) {
+      return 'yarn';
+    }
+    if (userAgent.startsWith('pnpm')) {
+      return 'pnpm';
+    }
+    if (userAgent.startsWith('npm')) {
+      return 'npm';
+    }
+  }
+  return 'yarn';
+}
+
+// From https://github.com/vercel/next.js/blob/canary/packages/create-next-app/helpers/is-folder-empty.ts
+
+async function isFolderEmpty(root: string, name: string): Promise<boolean> {
+  const files = await fs.readdir(root);
+
+  if (files.length > 0) {
+    // eslint-disable-next-line no-console
+    console.log(`The directory ${chalk.green(name)} is non-empty.`);
```

Comment:
Nit: does it make sense to move this logging to the consumer of `isFolderEmpty`? The function name doesn't suggest it will have side-effects.